lib/logstash/inputs/udp.rb

```diff
@@ -39,34 +39,17 @@ def initialize(params)
 
   public
   def register
-    @udp = nil
-  end # def register
-
-  public
-  def run(output_queue)
-  @output_queue = output_queue
-    begin
-      # udp server
-      udp_listener(output_queue)
-    rescue => e
-      @logger.warn("UDP listener died", :exception => e, :backtrace => e.backtrace)
-      Stud.stoppable_sleep(5) { stop? }
-      retry unless stop?
-    end # begin
-  end # def run
-
-  private
-  def udp_listener(output_queue)
     @logger.info("Starting UDP listener", :address => "#{@host}:#{@port}")
 
-    if @udp && ! @udp.closed?
-      @udp.close
-    end
-
     @udp = UDPSocket.new(Socket::AF_INET)
```

Comment:
run should not be concerned with the state of the world before. if we're restarting the plugin then stop/close should be closed before register and they should ensure everything is OK